tests/integration/utils.py

```diff
@@ -93,6 +93,29 @@ def wait_for_complete_request(self, response):
 
         return response
 
+    def fetch_all_requests(self, request_json=None):
+        """
+        Fetch a list of requests from the Cachito API.
```

Comment:
Could you please document that if `page` is in the `request_json` that the method will not iterate through all the pages?

Reply:
I wrote is inside the `while` loop: 
```
        while request_url:
            resp = requests.get(request_url, qtimeout=15)
            resp.raise_for_status()
            all_items += resp.json()["items"]
            if "page" in request_json:
                break
            request_url = resp.json()["meta"]["next"]
```

I can see it was not so informative. Do you know the way to write it better? Thank you.

Reply:
Thank you!

@mprahl proposed changes in private chat:
```
 def fetch_all_requests(self, query_params=None, all_pages=True):
        """
        Fetch a list of requests from the Cachito API.

        :param dict query_params: Request parameters and values (page, per_page, status, verbose)
        :return: Object that contains response from the Cachito API
        :rtype: list
        """
        if not query_params:
            query_params = {}
        request_url = f"{self._cachito_api_url}/requests"
        all_items = []
        while request_url:
            resp = requests.get(request_url, params=query_params, timeout=15)
            resp.raise_for_status()
            all_items += resp.json()["items"]
            if not all_pages:
                break
            request_url = resp.json()["meta"]["next"]

        return Response({"items": all_items}, None, resp.status_code)
```

I like the idea of simplicity. But I can imagine the cons: 

 In case somebody put `query_params={page:1}` and forget to set `all_pages` flag, it will be a conflict. Of course, we can change `if`: 
```
if "page" in query_params or not all_pages:
    break
```

but it is overcomplicated to me. 

@lcarva I think, we need your opinion.